core/Plugin/Report.php

```diff
@@ -325,55 +291,29 @@ public function configureView(ViewDataTable $view)
      */
     public function render()
     {
+        $viewDataTable = Common::getRequestVar('viewDataTable', false, 'string');
+        $fixed = Common::getRequestVar('forceView', 0, 'int');
+
+        $module = $this->getModule();
+        $action = $this->getAction();
+
         $apiProxy = Proxy::getInstance();
 
-        if (!$apiProxy->isExistingApiAction($this->module, $this->action)) {
-            throw new Exception("Invalid action name '$this->action' for '$this->module' plugin.");
+        if (!$apiProxy->isExistingApiAction($module, $action)) {
+            throw new Exception("Invalid action name '$module' for '$action' plugin.");
         }
 
-        $apiAction = $apiProxy->buildApiActionName($this->module, $this->action);
-
-        $view = ViewDataTableFactory::build(null, $apiAction, $this->module . '.' . $this->action);
+        $apiAction = $apiProxy->buildApiActionName($module, $action);
 
-        $rendered  = $view->render();
+        $view = ViewDataTableFactory::build($viewDataTable, $apiAction, $module . '.' . $action, $fixed);
 
-        return $rendered;
+        return $view->render();
     }
 
-    /**
-     * By default a widget will be configured for this report if a {@link $widgetTitle} is set. If you want to customize
-     * the way the widget is added or modify any other behavior you can overwrite this method.
-     * @param WidgetsList $widget
-     * @api
-     */
-    public function configureWidget(WidgetsList $widget)
+    public function configureWidgets(WidgetsList $widgetsList, ReportWidgetFactory $factory)
     {
-        if ($this->widgetTitle) {
-            $params = array();
-            if (!empty($this->widgetParams) && is_array($this->widgetParams)) {
-                $params = $this->widgetParams;
-            }
-            $widget->add($this->category, $this->widgetTitle, $this->module, $this->action, $params);
-        }
-    }
-
-    /**
-     * By default a menu item will be added to the reporting menu if a {@link $menuTitle} is set. If you want to
-     * customize the way the item is added or modify any other behavior you can overwrite this method. For instance
-     * in case you need to add additional url properties beside module and action which are added by default.
-     * @param \Piwik\Menu\MenuReporting $menu
-     * @api
-     */
-    public function configureReportingMenu(MenuReporting $menu)
```

Comment:
`configureReportingMenu` is no longer needed since the reporting menu is based on the exposed widgets